loom/engine/bindings/loom/lmSQLite.cpp

```diff
@@ -32,13 +32,129 @@ using namespace LS;
 //SQLite Statement binding for Loomscript
 class Statement
 {
+private:
+ //  Connection c; //crashes the build? Not sure if its because it's only defined later in the file? Not sure about c++
 public:
-    sqlite3_stmt *statementHandle = NULL;
+    sqlite3_stmt *statementHandle;
+
+   // Statement::Statement(Connection *c)
+  //  {
+
+  //  }
 
     int getParameterCount()
     {
-        return 666;
+        int count = sqlite3_bind_parameter_count(statementHandle);
+       // int error = connection->getErrorCode();
+       // if(error != SQLITE_OK)
+      //  {
+      //      lmLogError(gSQLiteGroup, "Error BLAH BLAH BLAH: with message: %s", connection->getErrorMessage());
+      //  }
+        return count;
+    }
+
+    const char *getParameterName(int index)
+    {
+        return sqlite3_bind_parameter_name(statementHandle, index);
+    }
+
+    int getParameterIndex(const char* name)
+    {
+        return sqlite3_bind_parameter_index(statementHandle, name);
+    }
+
+    void bindDouble(int index, double value)
+    {
+        int i = sqlite3_bind_double(statementHandle, index, value);
+        if(i != SQLITE_OK)
+        {
+            lmLogError(gSQLiteGroup, "Error BLAH BLAH BLAH: with message: %s", "error:" + i);
+        }
+    }
+
+    void bindInt(int index, int value)
+    {
+        int i =sqlite3_bind_int(statementHandle, index, value);
+        if(i != SQLITE_OK)
+        {
+            lmLogError(gSQLiteGroup, "Error BLAH BLAH BLAH: with message: %s", "error:" + i);
+        }
+    }
+
+    void bindBytes(int index, const sqlite3_value *value)
```

Comment:
I'm not sure what a ByteArray will be in C++ from Loomscript... I'll check this out.